plugins/modules/omevv/omevv_firmware.py

```diff
@@ -0,0 +1,689 @@
+        else:
+            self.module.exit_json(msg=CHANGES_NOT_FOUND_MSG, changed=False)
+
+    def _get_target(self, target_list):
+        for target in target_list:
+            return target
+
+    def _get_host_id(self, vcenter_uuid, target):
+        service_tag = target['servicetag']
+        host = target['host']
+
+        if service_tag:
+            return self.omevv_info_obj.get_host_id(vcenter_uuid, hostname=None,
+                                                   servicetag=service_tag)
+        else:
+            return self.omevv_info_obj.get_host_id(vcenter_uuid, hostname=host, servicetag=None)
+
+    def _is_firmware_update_needed(self, vcenter_uuid, cluster_group_id, host_id, target):
+        before_dict = {}
+        after_dict = {}
+        diff_before = json.dumps(before_dict)
+        diff_after = json.dumps(after_dict)
+        firmware_drift_info = self.omevv_info_obj.get_firmware_drift_info_for_single_host(
+            vcenter_uuid, cluster_group_id, host_id)
+        current_firmware_components = firmware_drift_info.get(
+            'hostComplianceReports', [])[0].get('componentCompliances', [])
+
+        required_firmware_components = target['firmware_components']
+
+        firmware_update_needed = False
+        before_dict = {}
+        after_dict = {}
+
+        for component in required_firmware_components:
+            for current_component in current_firmware_components:
+                if current_component['sourceName'] == component and current_component['driftStatus'] == 'NonCompliant':
+                    before_dict[current_component["sourceName"]] = {"firmwareversion": current_component["currentValue"]}
+                    after_dict[current_component["sourceName"]] = {"firmwareversion": current_component["baselineValue"]}
+                    firmware_update_needed = True
+
+        if firmware_update_needed:
+            diff_before = before_dict
+            diff_after = after_dict
+
+        return firmware_update_needed, diff_before, diff_after
+
+    def _is_update_job_allowed(self, vcenter_uuid, cluster_group_id, cluster_name):
+        update_job_status = self.omevv_update_obj.check_existing_update_job(vcenter_uuid,
+                                                                            cluster_group_id)
+        if update_job_status is not True:
+            self.module.exit_json(msg=UPDATE_JOB_PRESENT_MSG.format(
+                group_id=cluster_group_id, cluster_name=cluster_name), skipped=True)
+            return False
+        return True
+
+    def _is_job_name_existing(self, vcenter_uuid, job_name):
+        job_exist_status = self.omevv_update_obj.check_existing_job_name(vcenter_uuid, job_name)
+        if job_exist_status is True:
+            self.module.exit_json(msg=JOB_NAME_ALREADY_EXISTS_MSG.format(
+                job_name=job_name), failed=True)
+            return True
```

Comment:
exit_json will end the module execution, we can remove "return True"

Reply:
Incorporated